### PR TITLE
♻️ fixes mypy issues in `models-library`

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- 🔨 adding mypy check for models-library in the CI
- ♻️ fixes mypy issues in models-library 

## Related issue/s
- related to #4071
- closes https://github.com/ITISFoundation/osparc-simcore/issues/4400

## How to test
```cmd
cd packages/models-library
make install-dev
make mypy
```

### CODE REVIEW

File: packages/models-library/src/models_library/aiodocker_api.py [hunk 1]
```diff
@@ -55,6 +55,6 @@ class AioDockerTaskSpec(TaskSpec):
 class AioDockerServiceSpec(ServiceSpec):
     TaskTemplate: AioDockerTaskSpec | None = None
 
-    class Config(ServiceSpec.Config):
```
Comment (pcrespov) [inline packages/models-library/src/models_library/aiodocker_api.py:58]: Careful with this change.
NOTE that the `ServiceSpec.Config` does not get automatically inherited and it needs to be done this way.

File: packages/models-library/src/models_library/aiodocker_api.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 
 class AioDockerContainerSpec(ContainerSpec):
-    Env: dict[str, str | None] | None = Field(
+    Env: dict[str, str | None] | None = Field(  # type: ignore
```
Comment (pcrespov) [inline packages/models-library/src/models_library/aiodocker_api.py:14]: why this ignore? try `Field(default=None, description="some description")` !

File: packages/models-library/src/models_library/api_schemas_storage.py [hunk 3]
```diff
@@ -131,7 +131,9 @@ class FileMetaDataGet(BaseModel):
     )
     created_at: datetime
     last_modified: datetime
-    file_size: ByteSize = Field(-1, description="File size in bytes (-1 means invalid)")
+    file_size: ByteSize | int = Field(
+        -1, description="File size in bytes (-1 means invalid)"
```
Comment (pcrespov) [inline packages/models-library/src/models_library/api_schemas_storage.py:135]: Use key arguments `default=-1`

**Review by pcrespov — CHANGES_REQUESTED** [review]
thx!
Please double check `Config` inheritance

**Review by sanderegg — APPROVED** [review]

File: packages/models-library/src/models_library/aiodocker_api.py (hunk 2 above)
Comment (matusdrobuliak66) [reply packages/models-library/src/models_library/aiodocker_api.py:14]: `Incompatible types in assignment (expression has type "Optional[Dict[str, Optional[str]]]", base class "ContainerSpec" defined the type as "Optional[List[str]]")  [assignment]`  I guess thats the reason why we have `convert_list_to_dict` validator

File: packages/models-library/src/models_library/aiodocker_api.py (hunk 1 above)
Comment (matusdrobuliak66) [reply packages/models-library/src/models_library/aiodocker_api.py:58]: You are right, returned back with `# type: ignore`. In Pydantic v2 there will be better ways how to approach this.

**Review by pcrespov — APPROVED** [review]
:tada:  we are officially mypyfied

**Pull request merged**